subsys/fs/littlefs_fs.c

```diff
@@ -181,14 +181,49 @@ static void release_file_data(struct fs_file_t *fp)
 	fp->filep = NULL;
 }
 
-static int littlefs_open(struct fs_file_t *fp, const char *path)
+static int lfs_flags_from_zephyr(unsigned int zflags)
+{
+	int flags = 0;
+
+	if (zflags & FS_O_CREATE) {
+		flags |= LFS_O_CREAT;
+		zflags &= ~FS_O_CREATE;
+	}
+
+	/* LFS_O_READONLY and LFS_O_WRONLY can be selected at the same time,
+	 * this is not a mistake, together they create RDWR access.
+	 */
+	if (zflags & FS_O_READ) {
+		flags |= LFS_O_RDONLY;
+		zflags &= ~FS_O_READ;
+	}
+
+	if (zflags & FS_O_WRITE) {
+		flags |= LFS_O_WRONLY;
+		zflags &= ~FS_O_WRITE;
+	}
+
+	if (zflags & FS_O_APPEND) {
+		flags |= LFS_O_APPEND;
+		zflags &= ~FS_O_APPEND;
+	}
+
+	if (zflags != 0) {
```

Comment:
I don't think it's worth emitting a warning for something that causes no harm.  This will only be a problem if/when we add more flags and don't update this module.  It would be reasonable to just get rid of this, and so also all the actions that clear the individual bits of `zflags` (which is also a little weird).

Supporting this perspective: you didn't check for extra flags in the fat_fs translation.  So please remove it here.

Reply:
Done.